### DISCUSSION TITLE
Nuxt layers with workspaces

### DISCUSSION THREAD

**Discussion by igorjacauna** [nuxt/nuxt#15967] [Questions] [answered]
What I'm doing wrong with this repo? https://github.com/igorjacauna/nuxt-layers-yarn-workspace

`nuxt-b` extends `nuxt-a` package on this workspace but when run `nuxt-b` (`yarn dev`) shows `Cannot extend config from nuxt-a`

There is a script on workspace root that runs `yarn workspace nuxt-b dev`

**Comment by IsraelOrtuno** [comment]
Could you try to add `nuxt-a` as a dependence to `package.json` in `nuxt-b`?

```json
  "dependencies": {
    "nuxt-a": "*"
  }
```

Then run `yarn` from your the root of the project and `yarn dev` from `nuxt-b` folder

**Reply by igorjacauna** [reply]
Thanks for help! 

You mean instead in `devDependecies` as it is [here](https://github.com/igorjacauna/nuxt-layers-yarn-workspace/blob/main/packages/nuxt-b/package.json#L13)?

**Reply by igorjacauna** [reply]
Did not work adding `nuxt-a` to `depedencies` instead `devDepedencies` in `nuxt-b` `package.json`.

**Reply by IsraelOrtuno** [reply]
Yep! Didn't see it there. Not sure if moving it to `dependencies` will make any change since I don't see any reference to `nuxt-a` in the `yarn.lock` file. I kind of use this setup and pretty similar to yours so trying to find the differences.

**Reply by IsraelOrtuno** [reply]
In my case, using Yarn 3 (but was also working with yarn 1) and the yarn.lock is including my shared project

```
"@jobbb/base@*, @jobbb/base@workspace:packages/base":
```

**Reply by igorjacauna** [reply]
yarn 1 does not add `nuxt-a` or `nuxt-b` on `yarn.lock`. Do you think this could be related?

**Comment by igorjacauna** [comment]
I tried with `pnpm` too and same problem: `Cannot extend config from nuxt-a`

**Reply by yassilah** [reply]
Hey, I'm using pnpm with extends and it works great, make sure you have that in your package.json ;)

```json
  "exports": {
    ".": "./nuxt.config.ts"
  }
```

**Reply by igorjacauna** [reply]
That was the key! This worked even with `yarn`! Thanks!

**Comment by igorjacauna** [comment, marked as answer]
Adding this to `nuxt-a` `package.json` worked:

> ```json
>   "exports": {
>     ".": "./nuxt.config.ts"
>   }
> ```

Maybe this could be in docs?

**Reply by dsvgl** [reply]
@igorjacauna it's in the docs: https://nuxt.com/docs/guide/going-further/layers#npm-package
See second code block in that chapter. But it instead defines it via the `main` field:
`"main": "./nuxt.config.ts"`.
This is for cjs AFAIK. But it works with the "main" entry in my project with multiple packages, each being ESM (`"type": "module"`). 🤷🏻‍♂️

**Reply by igorjacauna** [reply]
Uow, thanks. Didn't see that.
